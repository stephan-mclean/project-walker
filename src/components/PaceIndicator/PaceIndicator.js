import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

const getTextPair = (content, indicator) => {
  return (
    <span>
      <span className="text--color-secondary">{content}</span>
      <span className="text--color-tertiary">{indicator}</span>
    </span>
  );
};

const PaceIndicator = ({
  hours,
  mins,
  timeIcon,
  distance,
  distanceMetric,
  distanceIcon,
  className
}) => {
  const paceClass = `high-contrast-coloring  ${className || ""}`;

  return (
    <div className={paceClass}>
      {(hours || mins) && (
        <div
          className="margin-right--xs display--inline-block"
          data-testid="timeContainer"
        >
          <FontAwesomeIcon icon={timeIcon} /> {hours && getTextPair(hours, "h")}
          {mins && getTextPair(mins, "m")}
        </div>
      )}

      {distance && (
        <div className="display--inline-block" data-testid="distanceContainer">
          <FontAwesomeIcon icon={distanceIcon} />{" "}
          {getTextPair(distance, distanceMetric)}
        </div>
      )}
    </div>
  );
};

PaceIndicator.propTypes = {
  hours: PropTypes.number,
  mins: PropTypes.number,
  timeIcon: PropTypes.string,
  distance: PropTypes.number,
  distanceMetric: PropTypes.string,
  className: PropTypes.string
};

PaceIndicator.defaultProps = {
  timeIcon: "stopwatch",
  distanceIcon: "map-marker"
};

export default PaceIndicator;
