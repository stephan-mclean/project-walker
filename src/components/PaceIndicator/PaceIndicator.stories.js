import React, { Fragment } from "react";
import { storiesOf } from "@storybook/react";

import PaceIndicator from "./PaceIndicator";

storiesOf("Components", module).add("PaceIndicator", () => {
  return (
    <Fragment>
      <PaceIndicator
        mins={32}
        distance={3.2}
        distanceMetric="km"
        className="margin-bottom"
      />

      <PaceIndicator
        hours={1}
        mins={32}
        distance={3.2}
        distanceMetric="km"
        className="high-contrast-coloring margin-bottom"
      />
    </Fragment>
  );
});
