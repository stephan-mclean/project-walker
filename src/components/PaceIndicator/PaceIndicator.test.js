import React from "react";
import { shallow } from "enzyme";

import PaceIndicator from "./PaceIndicator";

describe("Testing the PaceIndicator component", () => {
  it("Should render the time", () => {
    const indicator = shallow(<PaceIndicator hours={1} mins={32} />);

    expect(indicator).toMatchSnapshot();

    const timeContainer = indicator.find('div[data-testid="timeContainer"]');
    expect(timeContainer.exists()).toBeTruthy();

    expect(
      timeContainer
        .find("span")
        .first()
        .text()
    ).toEqual("1h");

    expect(
      timeContainer
        .find("span")
        .at(3)
        .text()
    ).toEqual("32m");
  });

  it("Should render the correct time icon", () => {
    const indicator = shallow(<PaceIndicator mins={32} timeIcon="someicon" />);

    expect(indicator).toMatchSnapshot();

    const timeContainer = indicator.find('div[data-testid="timeContainer"]');

    const icon = timeContainer.find("FontAwesomeIcon");
    expect(icon.prop("icon")).toEqual("someicon");
  });

  it("Should render the distance", () => {
    const indicator = shallow(
      <PaceIndicator distance={3.2} distanceMetric="km" />
    );

    expect(indicator).toMatchSnapshot();

    const distanceContainer = indicator.find(
      'div[data-testid="distanceContainer"]'
    );

    expect(
      distanceContainer
        .find("span")
        .first()
        .text()
    ).toEqual("3.2km");

    expect(distanceContainer.exists()).toBeTruthy();
  });

  it("Should render the correct distance icon", () => {
    const indicator = shallow(
      <PaceIndicator distance={3.2} distanceIcon="someicon" />
    );

    expect(indicator).toMatchSnapshot();

    const distanceContainer = indicator.find(
      'div[data-testid="distanceContainer"]'
    );

    const icon = distanceContainer.find("FontAwesomeIcon");
    expect(icon.prop("icon")).toEqual("someicon");
  });

  it("Should allow passing a custom class", () => {
    const indicator = shallow(
      <PaceIndicator mins={32} className="high-contrast-coloring" />
    );

    expect(indicator).toMatchSnapshot();

    expect(indicator.hasClass("high-contrast-coloring")).toBeTruthy();
  });
});
