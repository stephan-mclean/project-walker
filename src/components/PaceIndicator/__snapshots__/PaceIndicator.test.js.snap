// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the PaceIndicator component Should allow passing a custom class 1`] = `
<div
  className="default-coloring high-contrast-coloring"
>
  <div
    className="margin-right--xs display--inline-block"
    data-testid="timeContainer"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="stopwatch"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    <span>
      <span
        className="text--color-secondary"
      >
        32
      </span>
      <span
        className="text--color-tertiary"
      >
        m
      </span>
    </span>
  </div>
</div>
`;

exports[`Testing the PaceIndicator component Should render the correct distance icon 1`] = `
<div
  className="default-coloring "
>
  <div
    className="display--inline-block"
    data-testid="distanceContainer"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="someicon"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    <span>
      <span
        className="text--color-secondary"
      >
        3.2
      </span>
      <span
        className="text--color-tertiary"
      />
    </span>
  </div>
</div>
`;

exports[`Testing the PaceIndicator component Should render the correct time icon 1`] = `
<div
  className="default-coloring "
>
  <div
    className="margin-right--xs display--inline-block"
    data-testid="timeContainer"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="someicon"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    <span>
      <span
        className="text--color-secondary"
      >
        32
      </span>
      <span
        className="text--color-tertiary"
      >
        m
      </span>
    </span>
  </div>
</div>
`;

exports[`Testing the PaceIndicator component Should render the distance 1`] = `
<div
  className="default-coloring "
>
  <div
    className="display--inline-block"
    data-testid="distanceContainer"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="map-marker"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    <span>
      <span
        className="text--color-secondary"
      >
        3.2
      </span>
      <span
        className="text--color-tertiary"
      >
        km
      </span>
    </span>
  </div>
</div>
`;

exports[`Testing the PaceIndicator component Should render the time 1`] = `
<div
  className="default-coloring "
>
  <div
    className="margin-right--xs display--inline-block"
    data-testid="timeContainer"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="stopwatch"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    <span>
      <span
        className="text--color-secondary"
      >
        1
      </span>
      <span
        className="text--color-tertiary"
      >
        h
      </span>
    </span>
    <span>
      <span
        className="text--color-secondary"
      >
        32
      </span>
      <span
        className="text--color-tertiary"
      >
        m
      </span>
    </span>
  </div>
</div>
`;
